shared/subspace-data-retrieval/src/object_fetcher.rs

```diff
@@ -153,16 +166,29 @@ where
     /// Create a new object fetcher with the given configuration.
     ///
     /// `max_object_len` is the amount of data bytes we'll read for a single object before giving
-    /// up and returning an error, or `None` for no limit (`usize::MAX`).
+    /// up and returning an error. In this implementation, it is limited to
+    /// [`MAX_SUPPORTED_OBJECT_LENGTH`], which is much larger than the largest block on any domain
+    /// (as of December 2024).
     pub fn new(
         piece_getter: Arc<PG>,
         erasure_coding: ErasureCoding,
-        max_object_len: Option<usize>,
+        mut max_object_len: usize,
     ) -> Self {
+        if max_object_len > MAX_SUPPORTED_OBJECT_LENGTH {
+            warn!(
+                max_object_len,
+                MAX_SUPPORTED_OBJECT_LENGTH,
+                "Object fetcher size limit exceeds maximum supported object size, \
+                limiting to implementation-supported size"
+            );
+
+            max_object_len = MAX_SUPPORTED_OBJECT_LENGTH;
+        }
```

Comment:
The current implementation assumes the largest compact encoded object length is 4 bytes, or a `1 GB - 1` length:
https://github.com/autonomys/subspace/blob/b856ccebc682881128532afce4eae62a0ef05802/shared/subspace-data-retrieval/src/object_fetcher.rs#L29-L34

We can change that assumption any time we like, but it seems pointless right now, because the largest current block size in any domain is 5 MB.